test/system/multifactor_auths_test.rb

```diff
@@ -17,6 +17,44 @@ class MultifactorAuthsTest < ApplicationSystemTestCase
     @user.disable_totp!
   end
 
+  test "setup mfa does not cache OTP setup" do
+    sign_in
+    visit edit_settings_path
+
+    click_button "Register a new device"
+    saved_otp_key = otp_key
+    totp = ROTP::TOTP.new(saved_otp_key)
+    fill_in "otp", with: totp.now
+    click_button "Enable"
+
+    assert page.has_content? "Recovery codes"
+
+    go_back
+
+    assert page.has_content? "has already been enabled"
+    refute page.has_content? "Register a new device"
+    refute page.has_content? saved_otp_key
```

Comment:
Just a note if we ever revisit this: I had trouble getting this test to fail without the filter. Even though I could make it go back in my browser, in the test it would do "has already been enabled". Since I couldn't get it to fail, I just left it here as an intention. The other test below _does_ fail without the filter.